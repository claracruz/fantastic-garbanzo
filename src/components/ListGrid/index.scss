.list-grid {
    list-style-type: none;
    padding: 0;
}
