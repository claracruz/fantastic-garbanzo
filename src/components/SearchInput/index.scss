
.wrap {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 20px;
    height: 65px;

    h2 {
        margin-right: 40px;
        position: relative;
        top: 3px;
    }

    .search {
        max-width: 320px;
        width: 100%;
        position: relative;
        margin-right: 35px;
    }

    .search-input {;
        width: 100%;
        border: 3px solid #0198A6;
        padding: 5px;
        height: 36px;
        border-radius: 5px;
        outline: none;
        font-size: 0.8125rem;
        background-color: white;
    }

    .searchButton {
        position: absolute;
        right: -32px;
        width: 40px;
        height: 36px;
        border: 1px solid #0198A6;
        background: #0198A6;
        text-align: center;
        color: #fff;
        border-top-right-radius: 5px;
        border-bottom-right-radius: 5px;
        cursor: pointer;
        font-size: 1rem;
    }
}


