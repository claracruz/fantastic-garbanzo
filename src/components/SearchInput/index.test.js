import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { SearchInput } from '../../components/SearchInput';

describe('<SearchInput />', () => {

	let props;

	describe('Rendering', () => {
		beforeEach(() => {
			props = {
				onSearch: jest.fn()
			};
		});

		it('renders without crashing', () => {
			const div = document.createElement('div');
			ReactDOM.render(<SearchInput {...props} />, div);
			ReactDOM.unmountComponentAtNode(div);
		});

		it('renders correctly', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const searchInput = wrapper.find('input');
			const h2 = wrapper.find('h2');
			expect(wrapper).toMatchSnapshot();
			expect(searchInput.length).toBe(1);
			expect(h2.length).toBe(1);
		});

	});

	describe('State handling', () => {
		beforeEach(() => {
			props = {
				onSearch: jest.fn()
			};
		});

		it ('renders with default state', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			expect(wrapper.instance().state).toEqual({ searchTerm: '' });
		});

		it('renders as expected when state has searchTerm', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const searchTerm = 'test';
			wrapper.instance().setState({ searchTerm });
			expect(wrapper).toMatchSnapshot();

			const searchInput = wrapper.find('input');
			expect(searchInput.props().value).toEqual(searchTerm);
		});
	});

	describe('Input events', () => {
		beforeEach(() => {
			props = {
				onSearch: jest.fn()
			};
		});

		it('can handle keypress event', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const instance = wrapper.instance();
			const searchTerm = 'search term';
			const eventObj = { currentTarget: { value: searchTerm } };

			wrapper.find('input').simulate('keyPress', eventObj);
			expect(instance.state.searchTerm).toEqual(searchTerm);
		});

		it('Can handle change event', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const instance = wrapper.instance();
			const searchTerm = 'search term';
			const eventObj = { currentTarget: { value: searchTerm } };

			wrapper.find('input').simulate('change', eventObj);
			expect(instance.state.searchTerm).toEqual(searchTerm);
		});

		it('can handle enter key press', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const instance = wrapper.instance();
			const input = wrapper.find('input');
			const searchTerm = 'search term';
			instance.setState({ searchTerm });
			input.simulate('keyDown', {
				key: 'Enter',
				currentTarget: { value: searchTerm }
			});
			expect(props.onSearch).toBeCalledWith(searchTerm);
		});

		it('should handle search button click as expected', () => {
			const wrapper = shallow(<SearchInput {...props} />);
			const input = wrapper.find('input');
			const button = wrapper.find('button');
			const searchTerm = 'search term';
			input.simulate('keyPress', {currentTarget: {value: searchTerm}});
			button.simulate('click');
			expect(props.onSearch).toBeCalledWith(searchTerm);
		});
	});

	describe('onSubmit()', () => {
		beforeEach(() => {
			props = {
				onSearch: jest.fn()
			};
		});
	});
});
