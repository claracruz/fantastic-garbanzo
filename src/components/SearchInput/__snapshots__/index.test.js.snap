// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> Rendering renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    onSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <label
            htmlFor="search"
          >
            Search Papers
          </label>
        </h2>,
        <div
          className="search"
        >
          <input
            className="search-input"
            id="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter a search term"
            type="text"
            value=""
          />
          <button
            className="searchButton"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>,
      ],
      "className": "wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            htmlFor="search"
          >
            Search Papers
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Papers",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Search Papers",
          "type": "label",
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search-input"
              id="search"
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter a search term"
              type="text"
              value=""
            />,
            <button
              className="searchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>,
          ],
          "className": "search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "id": "search",
              "onChange": [Function],
              "onKeyDown": [Function],
              "onKeyPress": [Function],
              "placeholder": "Enter a search term",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "searchButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                },
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <label
              htmlFor="search"
            >
              Search Papers
            </label>
          </h2>,
          <div
            className="search"
          >
            <input
              className="search-input"
              id="search"
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter a search term"
              type="text"
              value=""
            />
            <button
              className="searchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>,
        ],
        "className": "wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              htmlFor="search"
            >
              Search Papers
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search Papers",
              "htmlFor": "search",
            },
            "ref": null,
            "rendered": "Search Papers",
            "type": "label",
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search-input"
                id="search"
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                placeholder="Enter a search term"
                type="text"
                value=""
              />,
              <button
                className="searchButton"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>,
            ],
            "className": "search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-input",
                "id": "search",
                "onChange": [Function],
                "onKeyDown": [Function],
                "onKeyPress": [Function],
                "placeholder": "Enter a search term",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "searchButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<SearchInput /> State handling renders as expected when state has searchTerm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    onSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <label
            htmlFor="search"
          >
            Search Papers
          </label>
        </h2>,
        <div
          className="search"
        >
          <input
            className="search-input"
            id="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter a search term"
            type="text"
            value="test"
          />
          <button
            className="searchButton"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>,
      ],
      "className": "wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            htmlFor="search"
          >
            Search Papers
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Papers",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Search Papers",
          "type": "label",
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search-input"
              id="search"
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter a search term"
              type="text"
              value="test"
            />,
            <button
              className="searchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>,
          ],
          "className": "search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "id": "search",
              "onChange": [Function],
              "onKeyDown": [Function],
              "onKeyPress": [Function],
              "placeholder": "Enter a search term",
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "searchButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                },
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <label
              htmlFor="search"
            >
              Search Papers
            </label>
          </h2>,
          <div
            className="search"
          >
            <input
              className="search-input"
              id="search"
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter a search term"
              type="text"
              value="test"
            />
            <button
              className="searchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>,
        ],
        "className": "wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              htmlFor="search"
            >
              Search Papers
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search Papers",
              "htmlFor": "search",
            },
            "ref": null,
            "rendered": "Search Papers",
            "type": "label",
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search-input"
                id="search"
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                placeholder="Enter a search term"
                type="text"
                value="test"
              />,
              <button
                className="searchButton"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>,
            ],
            "className": "search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-input",
                "id": "search",
                "onChange": [Function],
                "onKeyDown": [Function],
                "onKeyPress": [Function],
                "placeholder": "Enter a search term",
                "type": "text",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "searchButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
